src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/ThrowsTaglet.java

```diff
@@ -25,274 +25,635 @@
+                        continue;
+                    }
+                    if (!alreadyDocumentedExceptions.add(exceptionType)) {
+                        continue;
+                    }
+                    for (ThrowsTree t : r.get().throwsTrees()) {
+                        outputAnExceptionTagDeeply(exceptionSection, t, r.get().method(), alreadyDocumentedExceptions, typeSubstitutions, writer, utils);
+                    }
+                }
+            }
+            // Step 3. List those exceptions from the `throws` clause for which
+            //         no documentation was found on Step 2.
+            for (TypeMirror e : substitutedExceptionTypes) {
+                if (!alreadyDocumentedExceptions.add(e)) {
+                    continue;
+                }
+                exceptionSection.beginEntry(e);
+                exceptionSection.endEntry();
+            }
+            assert alreadyDocumentedExceptions.containsAll(substitutedExceptionTypes);
+        } catch (Failure.ExceptionTypeNotFound f) {
+            var ch = utils.getCommentHelper(f.holder());
+            utils.configuration.getMessages().warning(ch.getDocTreePath(f.tag().getExceptionName()),
+                    "doclet.throws.reference_not_found");
+            // add bad entry to section
+        } catch (Failure.NotExceptionType f) {
+            var ch = utils.getCommentHelper(f.holder());
+            var name = diagnosticDescriptionOf(f.type());
+            utils.configuration.getMessages().warning(ch.getDocTreePath(f.tag().getExceptionName()),
+                    "doclet.throws.reference_bad_type", name);
+            // add bad entry to section
+        } catch (Failure.InvalidMarkup f) {
+
+        } catch (Failure.Unsupported f) {
+            var ch = utils.getCommentHelper(f.holder());
+            utils.configuration.getMessages().warning(ch.getDocTreePath(f.tag().getExceptionName()),
+                    "doclet.throwsInheritDocUnsupported");
+        }
+        return exceptionSection.build();
+    }
+
+    private static Map<ThrowsTree, ExecutableElement> toExceptionTags(ExecutableElement e, List<ThrowsTree> tags) {
+        return tags.stream()
+                .collect(Collectors.toMap(Function.identity(), t -> e, (e1, e2) -> {
+                    // there should be no equal exception tags, hence no merging is expected
+                    throw newAssertionError(e1, e2);
+                }, LinkedHashMap::new));
+    }
+
+    // - we need to hijack tag processing just for the "@throws + @inheritDoc"
+    //   combination (because one-to-many we need access to exception section
+    //   being written rather than individual exception description)
+    // - populates exception section, depth-first
+    // - content elements are composable, tags are not (because of the DocTreePath,
+    //   one cannot practically use a list of tags cherry-picked from different doc comments)
+
+    // This adds entries late; who starts an entry also ends it
```

Comment:
Going forward (not here and now) we might want to generalize this.   For example, given that `{@inheritDoc}` applies to just the body, and not all block tags, we might want to allow something like `@see {@inheritDoc}` to inherit all `@see` tags from its supertype, suggesting that the one-to-many relationship might need to be more general. (And no, I don't have any suggestion for selective `@see` inheritance.)